lib/dk_buttons_and_leds/Kconfig

```diff
@@ -12,12 +12,33 @@ if DK_LIBRARY
 
 config DK_LIBRARY_BUTTON_SCAN_INTERVAL
 	int "Scanning interval of buttons in milliseconds"
+	default 50 if DK_LIBRARY_BUTTON_NO_ISR
 	default 10
 
 config DK_LIBRARY_DYNAMIC_BUTTON_HANDLERS
 	bool "Enable the runtime assignable button handler API"
 	default y
 
+config DK_LIBRARY_BUTTON_NO_ISR
+	bool "Poll buttons unconditionally (no interrupts) [EXPERIMENTAL]"
+	select EXPERIMENTAL
+	help
+	  With this option disabled, the module periodically scans all the
+	  available buttons until no button is pressed. If no button is
+	  pressed, the module uses GPIO interrupts to detect the first button
+	  press. On the first button press, the module switches back to
+	  periodically scanning buttons.
+
+	  Enable this option to avoid relying on GPIO interrupts at all and
+	  to unconditionally scan all buttons periodically. Please note that
+	  the constant scanning activity increases the overall power
+	  consumption of the system.
+
+	  For example, the option must be set for the nRF54L15 PDK revision
```

Comment:
@mkapala-nordic We still have the mention of nRF54L15 in the Kconfig description, shouldn't this be updated as well?

Reply:
The information from changelog are already present in the two first paragraphs, this nRF54L-related info is the additional context that it can be used mainly as a workaround, which we would rather leave this note here if possible.
If you think it would be better to remove it please let me know.

Reply:
Just wanted to confirm this - I think the note is fine considering that other nRF54L-related PRs are going in. Plus that's the approach we've agreed on. I'd just suggest adding a proper board revision naming.
Rewording suggestion:
For example, in case the application uses Button 3 or 4, the option must be set for the following nRF54L15 PDK revisions:
- PCA10156, v0.2.0 AA0-ES2
- PCA10156, v0.2.0 AA0-ES3
- PCA10056, v0.2.1 AB0-ES5

These versions of the PDK have Buttons 3 and 4 connected to the GPIO port which does not support interrupts.

Reply:
applied